src/olympia/scanners/admin.py

```diff
@@ -455,35 +501,9 @@ def formatted_matched_rules(self, obj):
 
     formatted_matched_rules.short_description = 'Matched rules'
 
-    def formatted_matched_rules_with_files(
-        self, obj, template_name='formatted_matched_rules_with_files'
-    ):
-        files_by_matched_rules = obj.get_files_by_matched_rules()
-        info = obj.rule_model._meta.app_label, obj.rule_model._meta.model_name
-        rules = (
-            [obj.matched_rule]
-            if hasattr(obj, 'matched_rule')
-            else obj.matched_rules.all()
-        )
-
-        return render_to_string(
-            f'admin/scanners/scannerresult/{template_name}.html',
-            {
-                'rule_change_urlname': 'admin:%s_%s_change' % info,
-                'external_site_url': settings.EXTERNAL_SITE_URL,
-                'file_id': (obj.version.file.id if obj.version else None),
-                'matched_rules': [
-                    {
-                        'pk': rule.pk,
-                        'name': str(rule),
-                        'description': str(rule.description),
-                        'files': files_by_matched_rules[rule.name],
-                    }
-                    for rule in rules
-                ],
-                'addon_id': obj.version.addon.pk if obj.version else None,
-                'version_id': obj.version.pk if obj.version else None,
-            },
+    def formatted_matched_rules_with_files(self, obj, *, template_name=None):
```

Comment:
This function seems obsolete (any reference to it could be replaced by `formatted_matched_rules_with_files_and_data` directly (though maybe it needs a snappier name)

Reply:
It's there because it's referenced in the admin `fields`. I just realized we could use a callable instead though, so I'll try that.